### PR TITLE
[Enhancement] optimize the re-balance strategy for backend scan range

### PR DESCRIPTION
**Pull request by Youngwb**
Fixes #issue
optimize the re-balance strategy for backend scan range
**before:**
tpch q1:
img_v2_2c795289-f568-4f20-9306-d526dec37f6g

backend 1 re-balance bytes: 609M+
backend 2 re-balance bytes: 3.8G+
backend 3 re-balance bytes: 811M+

tpch q2:
image

For partsupp
backend 1 re-balance bytes:  536M+
backend 2 re-balance bytes: 1.2G+
backend 3 re-balance bytes: 0

**now :**
tpch q1:
img_v2_964962bb-3d29-4a8a-b579-b3bbe57ff43g

backend 1 re-balance bytes: 0
backend 2 re-balance bytes : 1.0G+
backend 3 re-balance bytes: 134M

tpch q2:
64c433c9-35e5-42fa-9d5b-c471d0a3f253

For partsupp
backend 1 re-balance bytes:  391M+
backend 2 re-balance bytes: 930M+
backend 3 re-balance bytes: 0

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 10 / 11 (90.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/RemoteScanRangeLocations.java|0|1|00.00%|[88]|
|:large_blue_circle:|com/starrocks/qe/HDFSBackendSelector.java|10|10|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/qe/HDFSBackendSelector.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ public class HDFSBackendSelector implements BackendSelector {
     private final boolean forceScheduleLocal;
     private final boolean shuffleScanRange;
     private final int kCandidateNumber = 3;
-    private final int kMaxImbalanceRatio = 3;
+    private final double kMaxImbalanceRatio = 1.1;
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/qe/HDFSBackendSelector.java:85]: Could you elaborate a little bit the reason of choosing this value

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/qe/HDFSBackendSelector.java:87]: Done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 10 / 11 (90.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/RemoteScanRangeLocations.java|0|1|00.00%|[88]|
|:large_blue_circle:|com/starrocks/qe/HDFSBackendSelector.java|10|10|100.00%|[]|

**Review by dirtysalt — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**